src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2Serializer.cs

```diff
@@ -199,12 +199,15 @@ protected virtual Saml2Advice ReadAdvice(XmlDictionaryReader reader)
         /// <returns>A <see cref="Saml2Assertion"/> instance.</returns>
         public virtual Saml2Assertion ReadAssertion(XmlReader reader)
         {
+            if (reader == null)
```

Comment:
For consistency with SamlSerializer, `if (reader == null)` check could be removed and `XmlUtil.CheckReaderOnEntry(reader, Saml2Constants.Elements.Assertion, Saml2Constants.Namespace);` line could be placed on the top of this method.

Reply:
We wanted to be explicit about not supporting encrypted assertions. For that we need the check below.